src/comments.js

```diff
@@ -412,19 +416,27 @@ function printComments(path, print, options) {
             parent
           )
         );
+      } else {
+        // Those two node types print their dangling comments themselves
+        if (value.type !== "Program" && value.type !== "BlockStatement") {
+          trailingParts.push(
+            printDanglingComment(
+              commentPath,
+              print,
+              options
+            )
+          );
+        }
```

Comment:
This looks cool, but I don't quite understand how it works?

Dangling comments exist when there is no node beside them to attach them to, so they are attached to the "parent" node the encapsulates them. This means they would be attached to the `ObjectExpression` for example if it's `{/* foo */}`. But in `ObjectExpression` we do:

```js
      if (props.length === 0) {
        return "{}";
      }
```

So there's literally no way to automatically print dangling comments. All expressions that can have dangling comments must manually print them themselves (there are probably a small number of places that can have them)

If you look at the JSX case in the tests, it must be printing the comment when it's printing the "expression" property of the `JSXExpressionContainer`. But what node is the "expression" property? It's an empty JSX expression block , so there are no nodes there. I'm curious how it automatically prints the dangling comment if there are no nodes there.

I'm probably missing something, just want to understand it :)